src/storage/src/source/reclock.rs

```diff
@@ -599,8 +599,25 @@ where
         }
     }
 
-    pub async fn compact(&mut self, new_since: Antichain<IntoTime>) {
-        self.remap_handle.compact(new_since).await;
+    /// Closes the collection from further input.
+    pub async fn finalize(&mut self) {
```

Comment:
I think we shouldn't have an explicit finalization method and instead automatically finalize the remap shard when the input frontier reaches the empty antichain

Reply:
Do you suggest we just do this in `ReclockOperator::mint`? We could overwrite the upper returned by the `Clock` to the empty antichain if we see that the source's upper is the empty antichain. If not here, where do you suggest we do this?

Reply:
Yes, exactly! If we know there no more changes we don't need to wait for ticks anymore

Reply:
Did this in [storage: seal remap shard input when source input closes](https://github.com/MaterializeInc/materialize/pull/15813/commits/6a2e069d95207e4ea0f33408e2b811961c948c4e) but moving the `return` condition felt a little odd--lmk if you had a different vision for this.